plugin/pkg/auth/authenticator/token/oidc/oidc.go

```diff
@@ -95,69 +126,85 @@ func New(opts OIDCOptions) (*OIDCAuthenticator, error) {
 		TLSClientConfig: &tls.Config{RootCAs: roots},
 	})
 
-	hc := &http.Client{}
-	hc.Transport = tr
+	return &OIDCAuthenticator{
+		issuerURL:       opts.IssuerURL,
+		trustedClientID: opts.ClientID,
+		usernameClaim:   opts.UsernameClaim,
+		groupsClaim:     opts.GroupsClaim,
+		httpClient:      &http.Client{Transport: tr},
+	}, nil
+}
 
-	maxRetries := opts.MaxRetries
-	if maxRetries < 0 {
-		maxRetries = DefaultRetries
-	}
-	retryBackoff := opts.RetryBackoff
-	if retryBackoff < 0 {
-		retryBackoff = DefaultBackoff
+// Close closes the OIDC authenticator, this will close the provider sync goroutine.
+func (a *OIDCAuthenticator) Close() {
+	a.mu.Lock()
+	defer a.mu.Unlock()
+
+	a.closed = true
+
+	if a.close != nil {
+		a.close()
 	}
+	return
+}
 
-	for i := 0; i <= maxRetries; i++ {
-		if i == maxRetries {
-			return nil, fmt.Errorf("failed to fetch provider config after %v retries", maxRetries)
-		}
+func (a *OIDCAuthenticator) client() (*oidc.Client, error) {
```

Comment:
This makes checking the `closed` flag harder because `atomic.Value` can never store a nil value. Once the authenticator is initialized, it'll never reach the check

```
if a.closed {
    return nil, errors.New("authenticator closed")
}
```

There are workarounds, such as shoving more values into the `atomic.Value` field.

Just going to nuke the `closed` field on the authorizer. The `Close()` method is mostly about stopping associated goroutines and is only ever called in tests. Don't think we actually need to enforce always returning an error after calling `Close()`.